### ISSUE TITLE
Incorrect result when casting Timestamp to Varchar 

### ISSUE THREAD

**Issue by kagamiori** [facebookincubator/velox#4169]
### Bug description

Casting Timestamp to Varchar is supported in Presto. 
```
presto:di> select cast(TIMESTAMP '1970-01-01 00:00:00' as varchar);
          _col0          
-------------------------
 1970-01-01 00:00:00.000 
(1 row)
```

This conversion is allowed in Velox, but not covered in unit tests. In addition, it produces incorrect results when I tried to add the test.

```
TEST_F(CastExprTest, timestamp) {
  testCast<Timestamp, std::string>(
      "string",
      {
          Timestamp(0, 0),
          std::nullopt,
      },
      {
          "1970-01-01 00:00:00",
          std::nullopt,
      });
}
```

### System information

N/A

### Relevant logs

```bash
[ RUN      ] CastExprTest.timestamp
velox/vector/tests/utils/VectorTestBase.cpp:123: Failure
Value of: expected->equalValueAt(actual.get(), i, i)
  Actual: false
Expected: true
at 0: expected 1970-01-01 00:00:00, but got 
[  FAILED  ] CastExprTest.timestamp (8 ms)
```
(Timestamp(0, 0) represents `TIMESTAMP '1970-01-01 00:00:00'`, so we expect the result to be a string "1970-01-01 00:00:00". But Velox returns an empty string.)

**Comment by singcha** [comment]
@mbasmanova - Can you assign this to me ?

**Comment by mbasmanova** [comment]
The empty result is due to Timestamp.h having no implementation for toAppend function:

```
template <typename T>
void toAppend(const ::facebook::velox::Timestamp& value, T* result) {
  // TODO Implement
}
```

The first step might be to change this to

```
template <typename T>
void toAppend(const ::facebook::velox::Timestamp& value, T* result) {
  result->append(value.toString());
}
```

**Comment by singcha** [comment]
> The empty result is due to Timestamp.h having no implementation for toAppend function:
> 
> ```
> template <typename T>
> void toAppend(const ::facebook::velox::Timestamp& value, T* result) {
>   // TODO Implement
> }
> ```
> 
> The first step might be to change this to
> 
> ```
> template <typename T>
> void toAppend(const ::facebook::velox::Timestamp& value, T* result) {
>   result->append(value.toString());
> }
> ```

Thanks Masha! I have implemented same flow here :https://github.com/facebookincubator/velox/pull/4365

There are differences wrt Presto though :
Currently we specify timestamp with 9 decimal places (1970-01-01T00:00:00.000000000) while presto does 3 (2023-03-20T17:07:21.012-07:00) with timezone information.

Do we have to ensure that it matches Presto ? or we follow Velox format and its upto different query engines to convert in the format they want.

**Comment by singcha** [comment]
> > The empty result is due to Timestamp.h having no implementation for toAppend function:
> > ```
> > template <typename T>
> > void toAppend(const ::facebook::velox::Timestamp& value, T* result) {
> >   // TODO Implement
> > }
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > The first step might be to change this to
> > ```
> > template <typename T>
> > void toAppend(const ::facebook::velox::Timestamp& value, T* result) {
> >   result->append(value.toString());
> > }
> > ```
> 
> Thanks Masha! I have implemented same flow here :#4365
> 
> There are differences wrt Presto though : Currently we specify timestamp with 9 decimal places (1970-01-01T00:00:00.000000000) while presto does 3 (2023-03-20T17:07:21.012-07:00) with timezone information.
> 
> Do we have to ensure that it matches Presto ? or we follow Velox format and its upto different query engines to convert in the format they want.

Adding our offline discussion below :
timezone - there are two types: Timestamp and Timestamp With Time Zone.

Out of the box Velox provides the following:

- a set of special forms (if, switch, and, or, cast, coalesce, try) that follow Presto semantics by default and allow for changing semantics via configuration flags; search for queryConfig in velox/expression/CastExpr.cpp

- a set of PrestoSQL functions

- a set of SparkSQL functions
